pkg/build/registry/build/strategy.go

```diff
@@ -110,25 +112,39 @@ type detailsStrategy struct {
 
 // Prepares a build for update by only allowing an update to build details.
 // For now, this is the Spec.Revision field
+// We now need to update the build.status.Reason and build.Status.Message
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	status := newBuild.Status
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status = status
 }
 
 // Validates that an update is valid by ensuring that no Revision exists and that it's not getting updated to blank
 func (detailsStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	errors := field.ErrorList{}
-	if oldBuild.Spec.Revision != nil {
-		// If there was already a revision, then return an error
-		errors = append(errors, field.Duplicate(field.NewPath("status", "revision"), oldBuild.Spec.Revision))
+	oldRevision := oldBuild.Spec.Revision
+	newRevision := newBuild.Spec.Revision
+	glog.V(0).Infof("newbuild: %#v", newRevision)
+	glog.V(0).Infof("oldBuild: %#v", oldRevision)
+
+	if newRevision == nil {
+		errors = append(errors, field.Invalid(field.NewPath("spec", "revision"), nil, "cannot set an empty revision in build spec"))
 	}
-	if newBuild.Spec.Revision == nil {
-		errors = append(errors, field.Invalid(field.NewPath("status", "revision"), nil, "cannot set an empty revision in build status"))
+	// NOTE: use reflect.DeepEqual to see if the old one is as the new one? and
+	// skip this step that causes issues with updating other fields on the build
+	if !reflect.DeepEqual(oldRevision, newRevision) && oldRevision != nil {
+		// If there was already a revision, then return an error
+		// NOTE: the returned error is ambigous, because there's no
+		// build.Status.Revision - it's Spec.Revision  revision is not contained within
+		// Status
+		// NOTE: with reflect.DeepEqual we don't actually need an error
```

Comment:
We should display an error, for an example, take a look at:
https://github.com/openshift/origin/blob/master/pkg/build/api/validation/validation.go#L44-L52
It helps to understand why we're failing validation